aea/connections/base.py

```diff
@@ -53,14 +58,15 @@ def __init__(self):
 class Connection(Component, ABC):
     """Abstract definition of a connection."""
 
+    connection_id = None  # type: PublicId
+
     def __init__(
         self,
-        configuration: Optional[ConnectionConfig] = None,
+        configuration: ConnectionConfig,
```

Comment:
@marcofavorito I think this needs to be a required argument

Reply:
yes that's why we check in the body of `__init__`. The problem is that we want it a keyword argument because we use `__init__(**kwargs)` extensively in other parts of the framework